### ISSUE TITLE
subaru was not supposed to fail

### ISSUE THREAD

**Issue by rusefillc** [rusefi/rusefi#6064]
i believe https://github.com/rusefi/rusefi/commit/7e1a5459a62e8ca04849c6c881a77ac78d167b17 was not supposed to have subary failing since rusefi_config.txt was already fixed

looks like subaru did not re-generate and fail

only once gen-config task has produced better https://github.com/rusefi/rusefi/commit/fb6dedc1c22537113f1221e625ff89ee3f1e1f8d did subaru not fail https://github.com/rusefi/rusefi/commit/892006b810baf52848ce6c978f055e39eca8e03c

**Comment by chuckwagoncomputing** [comment]
We have a funny series of issues.
1. This issue is caused by Make deciding to build the bootloader first, and the bootloader doesn't know it needs to build the config files first. We'll fix this by including rusefi_config.mk in the bootloader Makefile.
2. rusefi_config.mk calls gen_config_board.sh, which expects to be called from the firmware directory, but we're calling it from firmware/bootloader. We'll fix this by `cd`ing to firmware in gen_config_board.sh
3. Now our BOARD_DIR is wrong, because the bootloader Make instance needs it to be relative to firmware/bootloader, but gen_config_board needs it to be relative to firmware. Linux and cygwin has a tool (realpath) to get the correct relative path, but...
4. macOS does not. We could solve this issue by installing it and requiring users to install it, but this one is extra annoying in that the macOS version installs as `grealpath`, so we'd have to check for that everywhere we use it and alias it to `realpath`. Another solution would be to use absolute paths, but...
5. Something in the java is turning `/` into `\` on Windows. This results in `java.nio.file.NoSuchFileException: \cygdrive\d\a\rusefi\rusefi\firmware\config\boards\f407-discovery\prepend.txt`. I suppose this could be a new issue with the switch to cygwin.

**Comment by rusefillc** [comment]
@chuckwagoncomputing need help want to see that NoSuchFileException where is it exactly? So far I've looked at https://github.com/rusefi/rusefi/actions/runs/8058007692/job/22010135657

**Comment by chuckwagoncomputing** [comment]
https://github.com/chuckwagoncomputing/rusefi/actions/runs/8059487251/job/22013988308

**Comment by chuckwagoncomputing** [comment]
Unless the java thing is an easy fix, I'm leaning toward installing realpath on macOS. We already need to install mtools, zip, dosfstools, and flock. The check to alias `grealpath` to `realpath` can be done in one line.
`which realpath || (which grealpath && alias realpath='grealpath')`

**Comment by rusefillc** [comment]
Actually there is a different error ``cmp: tunerstudio/generated/signature_f407-discovery.txt: No such file or directory`` let me address under https://github.com/rusefi/rusefi/issues/6067

This one is fine.
```
((firingOrder == 18) || (firingOrder == 22)))}
Reading ../unit_tests\triggers.txt
Processing TRIGGERTYPE 1 10 TT_FORD_ASPIRE 0.00 1
id=1, count=10, name=TT_FORD_ASPIRE
onWheel TT_FORD_ASPIRE
```
